src/AutoRest.CSharp/Common/Input/ApiTypes.cs

```diff
@@ -98,5 +101,8 @@ public FormattableString GetHttpPipelineKeyCredentialString(string pipelineField
         public abstract ExtensibleSnippets ExtensibleSnippets { get; }
 
         public abstract string LicenseString { get; }
+        public abstract MethodBodyStatement GetMultipartFormDataRequestContentAddStatment(VariableReference multipartContent, ValueExpression content, ValueExpression name, ValueExpression? fileName, ValueExpression? contentType);
+        public abstract MethodBodyStatement GetMultipartFormDataRequestContentWriteToStatment(VariableReference multipartContent, ValueExpression stream, ValueExpression? cancellationToken);
+        public abstract MethodBodyStatement GetMultipartFormDataRequestContentWriteToAsyncStatment(VariableReference multipartContent, ValueExpression stream, ValueExpression? cancellationToken);
```

Comment:
multipartContent does not have to be a `VariableReference`.
```suggestion
        public abstract MethodBodyStatement GetMultipartFormDataRequestContentAddStatment(ValueExpression multipartContent, ValueExpression content, ValueExpression name, ValueExpression? fileName, ValueExpression? contentType);
        public abstract MethodBodyStatement GetMultipartFormDataRequestContentWriteToStatment(ValueExpression multipartContent, ValueExpression stream, ValueExpression? cancellationToken);
        public abstract MethodBodyStatement GetMultipartFormDataRequestContentWriteToAsyncStatment(ValueExpression multipartContent, ValueExpression stream, ValueExpression? cancellationToken);
```

Reply:
Hello @ArcturusZhang  Actually we enter the VariableReference, not typedValueExpression. Because we need to support both azure and non-azure, so in `WriteMultiParts` where whose method used, we cannot pass TypedValueExpression, instead `VariableReference`

Reply:
every expression inherits `ValueExpression`. I am suggesting `ValueExpression` - because it is not always we pass in a variable, we could pass in an expression such as `something.Content` <- this is a value expression.